### ISSUE TITLE
Include ability to query status of CheckButtons widget

### ISSUE THREAD

**Issue by DanHickstein** [matplotlib/matplotlib#5930]
Similar to the discussion of RadioButtons a few months ago (see [here](http://stackoverflow.com/questions/27784575/get-status-of-matplotlib-radio-button) and [here](https://github.com/matplotlib/matplotlib/pull/3971#issuecomment-69268762)), it seems like there should be an ability to check the status of the check boxes in the [CheckButtons](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/widgets.py#L475) widget. 

A fairly reasonable solution was proposed in this [SO post](http://stackoverflow.com/questions/17812938/retrieving-the-selected-values-from-a-checkbuttons-object-in-matplotlib) (answer from user "Grunby"), where you can just query the visibility of the lines in each checkbox. It would be fairly easy to implement this as a function within the CheckButtons class, so that users could do something like:

``` python
from matplotlib.widgets import CheckButtons
check = CheckButtons(ax1, ('a','b','c'), (True, False, True))
check.get_status()
```

where `get_active()` would return `(True, False, True)` in this case.

I would be happy to work on a PR. Any thoughts on exactly how this should be implemented or what the function should be called to query the status?

**Comment by WeatherGod** [comment]
I wouldn't be against an ability to find out the status, the entire point
of these widgets is that the callback would know what the change is. If you
are querying the object, then it seems like your event model is broken.
Note that this would also be true for RadioButtons.

**Comment by DanHickstein** [comment]
I see your point. But in many cases, you don't necessarily want any action to take place when a check button is changed. For example, I am woking on a GUI where there are many check buttons, text boxes, and radio buttons. They all do nothing when changed. It is only when the big "Run Simulation" button is pressed that I want to query the status of all of the parameters.

Of course, I could include many callback that modify variables to track of the value of each item as it is changed. However, it's much more straightforward if I can just get the current status of each widget.

**Comment by tacaswell** [comment]
On the other hand, we know the state (so we can draw it) and it seems annoying to make the users set up a _separate_ set of that that gets updated via callbacks rather that just giving a view of the state we hold.

**Comment by DanHickstein** [comment]
@tacaswell, yes, this nicely summarizes my view. 

Actually, the CheckButtons class doesn't directly hold the status of the button, it merely reverses the visibility of the lines (``widgets.py``, line 595):
```python
l1.set_visible(not l1.get_visible())
```

Since the user originally sets the status of each button through an array called `actives`, then perhaps it makes the most sense for the class to just store this variable during the [`__init__()`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/widgets.py#L501) function
```python
self.actives = actives
```
and then we can update the actives array whenever the button is clicked in the [`set_active()`](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/widgets.py#L580) function:
```python
def set_active(self, index):
        ...
        l1, l2 = self.lines[index]
        l1.set_visible(not l1.get_visible())
        l2.set_visible(not l2.get_visible())
        self.actives[index] = l1.get_visible() # This is the line I added
```
This would allow the user to query the status just by getting `myCheckButtons.actives`.

How does that sound?

**Comment by WeatherGod** [comment]
You would want a copy of that variable, since it is a mutable. And any
getters to that mutable should also be copies. Probably easier to just
compute it on the fly at each request.

**Comment by DanHickstein** [comment]
Compute on the fly at each request, eh?

So, something like adding a function to the CheckButtons class:

```python
def get_status():
    status_list = []
    for (l1,l2) in lines:
        status_list.append( l1.get_visible() )
    return status_list
```

**Comment by WeatherGod** [comment]
That'll work, although I would do that as a list comprehension.

**Comment by DanHickstein** [comment]
Oh, good idea. Something like this?
```python
def get_status():
    return [l1.get_visible() for (l1,l2) in lines]
```

**Comment by tacaswell** [comment]
@DanHickstein That looks good, can you throw that into a PR?

**Comment by QuLogic** [comment]
Fixed by #6018, I believe.

**Comment by DanHickstein** [comment]
Correct. Thanks for closing this!
